### PR TITLE
add header and footer to pages that are not sign in and sign up pages

### PR DESCRIPTION
**Pull request by SachiGoto**
# Description
By default, the Header and Footer will be included, unless users are visiting the signup or sign-in pages.

Fixes # (issue)

<div style="display: flex; flex-wrap: wrap;">
    Screenshot 2024-01-04 at 21:28:19
    Screenshot 2024-01-04 at 21:28:02
    Image 3
    Image 4
</div>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshot

https://github.com/tabi-memo/frontend/assets/70562492/c1d197d2-3065-431a-834c-3f4bff41172d

### CODE REVIEW

File: app/providers/session-provider.tsx [hunk 1]
```diff
@@ -19,7 +20,14 @@ export function SessionProvider({
...
-  return (
+  const pathname = usePathname()
+  return pathname === '/signin' || pathname === '/signup' ? (
     <SessionContext.Provider value={uuid}>{children}</SessionContext.Provider>
+  ) : (
+    <SessionContext.Provider value={uuid}>
+      <Header />
+      {children}
+      <Footer />
+    </SessionContext.Provider>
```
Comment (kanatagu) [inline app/providers/session-provider.tsx]: Please be careful when you merge as uuid does not use anymore from this PR.
https://github.com/tabi-memo/frontend/pull/24/files#diff-ca5f077b80b85c86fc27347274786c9e9d9198289384d9e28c8f35d2e5602de7

File: app/activity/[id]/page.tsx [hunk 2]
```diff
@@ -43,7 +42,6 @@ export default function ActivityDetails() {
 
   return (
     <>
```
Comment (kanatagu) [inline app/activity/[id]/page.tsx]: You don't need React fragment as well! Please check with the other pages too

File: app/providers/session-provider.tsx [hunk 3]
```diff
@@ -19,7 +20,14 @@ export function SessionProvider({
   children: ReactNode
   uuid: string
 }) {
-  return (
+  const pathname = usePathname()
+  return pathname === '/signin' || pathname === '/signup' ? (
```
Comment (kanatagu) [inline app/providers/session-provider.tsx]: It's better to have these URL as constants valuable. In the future, there might be password reset and register pages.

Like this

`const AUTH_PATH_NAMES = ['/signin', '/signup']`

**Review by kanatagu — COMMENTED** [review]
I think Account page needs to be modified as well!

I removed the layout.tsx under account page!

File: app/providers/session-provider.tsx (hunk 1 above)
Comment (SachiGoto) [reply app/providers/session-provider.tsx]: Okay! I changed the variable from uuid to userId

File: app/providers/session-provider.tsx (hunk 3 above)
Comment (SachiGoto) [reply app/providers/session-provider.tsx]: Okay, that's really a good point! I added a variable AUTH_PATH_NAMES and use it for the logic. 
` const pathname = usePathname()
  const AUTH_PATH_NAMES = ['/signin', '/signup']`
`  return AUTH_PATH_NAMES.includes(pathname) ? ...`

File: app/account/page.tsx [hunk 4]
```diff
@@ -38,7 +38,9 @@ export default function AccountPage() {
   return (
     <>
       {loading || !data?.usersCollection ? (
-        <Loading />
+        <Flex minH="100vh" align="center" justify="center">
+          <Loading />
+        </Flex>
```
Comment (SachiGoto) [inline app/account/page.tsx:43]: I added min height so when the page is loading, the spinner is located in the middle. 
<table>
  <tr>
    <td valign="top">Screenshot 1</td>
    <td valign="top">Screenshot 2</td>
  </tr>
</table>

File: app/activity/[id]/page.tsx (hunk 2 above)
Comment (SachiGoto) [reply app/activity/[id]/page.tsx]: I removed unnecessary fragments!

File: app/providers/session-provider.tsx [hunk 5]
```diff
@@ -19,7 +20,16 @@ export function SessionProvider({
   children: ReactNode
   userId: string
 }) {
-  return (
+  const pathname = usePathname()
+  console.log(pathname)
```
Comment (kanatagu) [inline app/providers/session-provider.tsx:24]: Could you delete this?

**Review by kanatagu — APPROVED** [review]

**Review by samuraikun — APPROVED** [review]

**Review by YoshinariYamanaka — APPROVED** [review]

**Pull request merged**